src/backlight/portfolio/portfolio.py

```diff
@@ -90,14 +90,98 @@ def construct_portfolio(
     return Portfolio(positions)
 
 
-def calculate_pl(pt: Portfolio, mkt: List[MarketData]) -> pd.DataFrame:
+# def calculate_pl(pt: Portfolio, mkt: List[MarketData]) -> pd.DataFrame:
+#     """
+#     Apply the sum on the homogenized portfolio
+#     args:
+#         - portfolio : a defined portfolio
+#         - mkt : list of marketdata for each asset
+#         - base_ccy : asset of reference
+
+#     """
+#     pl = pt.value()
+#     return pl.sum(axis=1)
+
+
+def homogenize_pl(
+    pt: Portfolio, mkt: List[MarketData], base_ccy: str = "USD"
+) -> Portfolio:
+    """
+    Normalize PL to one asset reference and sum
+    args:
+       - portfolio : a defined portfolio
+       - mkt : list of marketdata for each asset
+       - base_ccy : asset of reference, all assets are converted this one
+    """
+    new_positions = []
+
+    for position in pt._positions:
+        ccy = position.symbol[-3:]
+        if ccy == base_ccy:
+            new_positions.append(
+                position.loc[mkt[0].index.intersection(pt._positions[0].index)].copy()
+            )
+        else:
+            new_positions.append(convert_positions(position, mkt, ccy, base_ccy))
+
+    return Portfolio(new_positions)
+
+
+def convert_positions(
+    position: Positions, mkt: List[MarketData], ccy: str, base_ccy: str
+) -> Positions:
+    """
+    Convert the values of a position in a different currency from MarketData
+    args:
+        - position : the position to convert
+        - mkt : market forex datas
+        - ccy : the currency of the position
+        - base_ccy : the currency to express the position in
+    """
+
+    ratios = get_ratios(mkt, ccy, base_ccy)
+    idx = pd.to_datetime(mkt[0].index.intersection(position.index))
+    pos_values = position.loc[idx].values
+    ratios_values = ratios.loc[idx].values.reshape(ratios.loc[idx].values.size, 1)
+
+    new_p = Positions(
+        pd.DataFrame(
+            data=pos_values * ratios_values, columns=position.columns, index=idx
+        )
+    )
+    new_p.symbol = position.symbol
+    return new_p
+
+
+def get_ratios(mkt: List[MarketData], ccy: str, base_ccy: str) -> pd.Series:
+    """
+    Get the ratios of ccy expressed in base_ccy depending on market datas
+    args:
+        - market : market forex datas
+        - ccy : the currency to convert from
+        - base_ccy : the currency to convert to
+    """
+    for market in mkt:
+        if ccy + base_ccy == market.symbol:
+            ratios = pd.Series(market.bid.values, index=market.index, dtype=float)
+        elif base_ccy + ccy == market.symbol:
+            ratios = pd.Series(market.ask.values, index=market.index, dtype=float)
+            ratios = ratios.apply(lambda x: 0 if x == 0 else 1.0 / float(x))
```

Comment:
You do not need to use ask/bid here. Just using mid.

If you want to use ask/bid precisely here, we need to change ask/bid by current position size.